.github/workflows/ci.yml

```diff
@@ -0,0 +1,36 @@
+name: CI
+
+on: [pull_request] # TODO: also `master`
+
+env:
+  IMAGE_TAG: github.sha
+
+jobs:
+
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Build Docker images
+      run: |
+        set -e
+        TARGET=prod make build
+        TARGET=dev make build
+
+    - name: Smoke tests
+      run: |
+        set -e
+        TARGET=prod .travis/smoke-test.sh
+        TARGET=dev .travis/smoke-test.sh
+
+    - name: Lint
+      run: make lint
+
+    - name: Test
```

Comment:
downstream steps are not executed if a previous one fails, at the moment

Reply:
What's the cost of separate actions? Multiple Docker builds?

Reply:
Seems multiple jobs, which start in isolated environments and hence Docker builds; unless the image can be shared as an artifact which is possible in principle but likely convoluted. Jobs enable parallelization as they can run on separate environments (actual VMs?)

We may also runn stuff in parallel at the microlevel (`make`, `xargs`) but they will be seen as a single step so doesn't make the failures clear. In general there are some dependencies between these steps, if the smoke tests do not run also `analyse.sh` likely won't (besides dependencies towards `make build` which previously was separated in Travis CI's `install` section).